src/Spice86.Core/Emulator/CPU/Registers/UInt32RegistersIndexer.cs

```diff
@@ -0,0 +1,19 @@
+namespace Spice86.Core.Emulator.CPU.Registers;
+
+using Spice86.Core.Emulator.Memory.ReaderWriter;
+
+/// <summary>
+/// Accesses 32-bit registers based on a given index.
+/// </summary>
+public class UInt32RegistersIndexer : RegistersIndexer<uint> {
+    private readonly IUIntReaderWriter _uIntArrayReaderWriter;
+
+    public UInt32RegistersIndexer(IUIntReaderWriter uIntArrayReaderWriter) {
+        _uIntArrayReaderWriter = uIntArrayReaderWriter;
+    }
+
+    public override uint this[uint index] {
+        get => _uIntArrayReaderWriter[(uint)index];
+        set => _uIntArrayReaderWriter[(uint)index] = value;
```

Comment:
Isn't index already an uint?

Reply:
indeed, there are lot of places like that in the code, Indexer actually accepts both uint and int